File: tests/fixtures/sample-project/services/user-service/src/services/user.service.ts
import { UserRepository } from '../repositories/user.repository';

const repo = new UserRepository();

export class UserService {
  async getAll() {
    return repo.findAll();
  }

  async getById(id: string) {
    return repo.findById(id);
  }

  async create(data: { name: string; email: string }) {
    return repo.create(data);
  }

  async delete(id: string) {
    return repo.delete(id);
  }
}

File: tests/fixtures/sample-python-project-negative/services/user_service/services/user_service.py
"""User service business logic."""


class UserService:
    def __init__(self, repository):
        self.repository = repository

    def get_user(self, user_id):
        return self.repository.find_by_id(user_id)

    def create_user(self, name, email):
        from ..models.user_model import User
        user = User(id=0, name=name, email=email)
        return self.repository.create(user)
